### PR TITLE
AVL trees

### PR DESCRIPTION
**Pull request by vgvoleg**
Signed-off-by: Oleg Ovcharuk <vgvoleg@gmail.com>

### CODE REVIEW

File: src/avl/CMakeLists.txt [hunk 1]
```diff
@@ -12,13 +12,13 @@ set(target ${LIBRARY})
 file(GLOB hrcs "AVL.h")
 file(GLOB srcs "AVL.cpp")
```
Comment (Lenferd) [inline src/avl/CMakeLists.txt]:
```suggestion
file(GLOB srcs "AVL.cpp" "AVLNode.cpp")
```
or for all files with extension 

```suggestion
file(GLOB srcs "*.cpp")
```

same below

File: src/avl/AVL.cpp [hunk 2]
```diff
@@ -2,13 +2,85 @@
...
 AVLTree::~AVLTree(){
 
 }
 
 void AVLTree::insert(const int value) {
-    usleep(value * 1000);
+    if (head) {
+        insert(head, value);
+    } else {
+        head = new AVLNode(value);
```
Comment (Lenferd) [inline src/avl/AVL.cpp]: Should we then destoy it in destructor?

File: src/avl/AVL.h [hunk 3]
```diff
@@ -5,19 +5,29 @@
...
+#include "AVLNode.h"
 
 class AVLTree {
 public:
     AVLTree();
     ~AVLTree();
 
     void insert(const int value);
-    void erase(const int value);
+    void remove(const int value);
```
Comment (Lenferd) [inline src/avl/AVL.h]: Maybe it's better to stay with same functions name, as in std?
https://en.cppreference.com/w/cpp/container/set

File: src/avl/CMakeLists.txt (hunk 1 above)
Comment (vgvoleg) [reply src/avl/CMakeLists.txt]: done

File: src/avl/AVL.h (hunk 3 above)
Comment (vgvoleg) [reply src/avl/AVL.h]: Я подумал что ты заюзал слово erase потому что delete нельзя)))
Исправлено

File: src/avl/AVL.cpp (hunk 2 above)
Comment (vgvoleg) [reply src/avl/AVL.cpp]: done, but review is still needed :)

**Review by Lenferd — APPROVED** [review]

**Pull request merged**